### ISSUE TITLE
Proxy doesn't work

### ISSUE THREAD

**Issue by A-7666** [celzero/rethink-app#1098]
Was really looking forward to being able to route the traffic to Tor/localhost:9050.
And finally Rethink got this feature.

However, it doesn't work.

Proxy -> Orbot:
the "Setup Orbot" entry is grayed out.

Proxy -> Other -> Setup SOCKS5 Proxy:
- App: None (grayed out)
- Hostname: 127.0.0.1
- Port: 9050
- pressing Set.
Result: nothing on the phone can connect to the internet.

Could you make the proxy work?

I'm running GrapheneOS / AOSP 13.

**Comment by ignoramous** [comment]
> Could you make the proxy work?

Install Orbot, start it, then make sure it is setup to accept incoming SOCKS5 over `9050`.

> And finally Rethink got this feature.

Rethink has had this feature for 3 years now.

**Comment by ignoramous** [comment]
Cannot forward connections to another app if VPN is in lockdown mode ("Block connections without VPN" is turned ON).

**Comment by ignoramous** [comment]
Yep. This is an Android limitation. Android won't let Orbot connect if Rethink is put in *VPN Lockdown* mode.

> some programs and system components access internet while bypassing Rethink

Unless *Configure* -> *Network* -> *Enable network visibility* is turned ON, *installed* apps cannot bypass the VPN tunnel Rethink setup *regardless* of whether it is in *VPN Lockdown* mode or not.

System apps may (not that they do) bypass *any* VPN tunnel regardless of whether the VPN is *Lockdown*. See: #224

**Comment by ignoramous** [comment]
> In this case what is the point of this setting at all?

This setting is for the *end user* to ensure that no *installed* app can bypass the VPN tunnel even if the VPN tunnel implementation had provisions to let apps do so. For example, apps like Blokada, Intra (and perhaps NetGuard?) may let *installed* apps bypass their tunnels. The *VPN Lockdown* ensures that those apps can't despite that.

As far as Rethink is concerned, it doesn't let any apps bypass by default. *Enable network visibility* and *Exclude* are ways for users to explicitly instruct Rethink to let certain apps through. And both these settings won't be honoured by Rethink if VPN is put in *Lockdown*.

**Comment by ignoramous** [comment]
For the end user, HTTP Connect and TCP-only SOCKS5 tunnels into Orbot are one and the same. It is but an implementation detail. Since Rethink supports both, it exposes both.

**Comment by ignoramous** [comment]
*Always-on VPN* is enough to start any VPN app on reboots. Rethink had a bug in the previous version where it wouldn't start despite being set to *Always-on*, but I think it's fixed in `v055a`.

But like you say, *VPN Lockdown* offers much better protection, for when the VPN app may crash or not start, *installed* apps won't be able to connect.

**Comment by ignoramous** [comment]
>  Functioning traffic routing to Tor with no leakage brought by VPN Lockdown - isn't it worth it? :-)

Hopefully network namespaces work, and this isn't required.
